### DISCUSSION TITLE
Tasmota often restart reason Hardware Watchdog

### DISCUSSION THREAD

**Discussion by vinceg0267** [arendst/Tasmota#19942] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
In two days of service I noticed 68 reboots with hardware watchdog. I don't know how to find the missing information.
Program version: 13.2.0(tasmota)
Core/SDK version 2_7_4_9/2.2.2-dev(38a443e)

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Wemos D1 mini, PZem 004T v3_____
- [x] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
STATUS = {"Status":{"Module":18,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota_B6C53F","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-IT.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:19:21","StartupUTC":"2023-11-06T04:59:49","Sleep":50,"CfgHolder":4617,"BootCount":68,"BCResetTime":"2023-10-22T20:53:33","SaveCount":115,"SaveAddress":"F7000"}}
STATUS2 = {"StatusFWR":{"Version":"13.2.0(tasmota)","BuildDateTime":"2023-10-19T09:07:34","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"372/699"}}
STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["FASTWEB-7HpvD6-2.4GHZ",""],"TelePeriod":10,"Resolution":"558180C0","SetOption":["00008009","2805C80000000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
STATUS4 = {"StatusMEM":{"ProgramSize":637,"Free":364,"Heap":22,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000410","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62,68","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
STATUS5 = {"StatusNET":{"Hostname":"tasmota-B6C53F-1343","IPAddress":"192.168.1.63","Gateway":"192.168.1.254","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.254","DNSServer2":"0.0.0.0","Mac":"7C:87:CE:B6:C5:3F","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.69","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_B6C53F","MqttUser":"mosquitto","MqttCount":2,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
STATUS7 = {"StatusTIM":{"UTC":"2023-11-06T05:19:10","Local":"2023-11-06T06:19:10","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"07:44","Sunset":"17:22"}}
STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
STATUS10 = {"StatusSNS":{"Time":"2023-11-06T06:19:10","ENERGY":{"TotalStartTime":"2023-10-22T21:45:01","Total":3.153,"Yesterday":1.752,"Today":0.172,"Period":[0,0,0],"Power":[0,0,2],"ApparentPower":[0,0,5],"ReactivePower":[0,0,5],"Factor":[0.00,0.00,0.34],"Frequency":[50,50,50],"Voltage":[240,240,240],"Current":[0.000,0.000,0.021]}}}
STATUS11 = {"StatusSTS":{"Time":"2023-11-06T06:19:10","Uptime":"0T00:19:21","UptimeSec":1161,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":2,"Wifi":{"AP":1,"SSId":"FASTWEB-7HpvD6-2.4GHZ","BSSId":"CC:B1:82:1F:4F:F8","Channel":1,"Mode":"11n","RSSI":100,"Signal":-29,"LinkCount":1,"Downtime":"0T00:00:05"}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment, marked as answer]
Known problem of the Wemos D1 mini clones. The onboard 3.3V Voltage regulator is to weak.
The orig Wemos D1 minis do have an 500mA Type. Nearly all so called Wemos D1 minis are fake/clones and do have only a 150mA Type. This is causing issues. To be clear, NOT an Tasmota problem.
Solving: Replace the Voltage regulator against a 500mA type or use a NodeMCU board.
NodeMCU boards (even the clones) have a decent 3.3V Voltage regulator.

**Reply by vinceg0267** [reply]
Thank's for your reply. Can you tell me how I can replace the voltage regulator or link me a  NodeMCU board?

**Reply by stefanbode** [reply]
Replacement is a quite tricky operation because it is very tiny on the board. But you can easily install a second one between the 5V, 3.3V and ground connector. This works very well. You might order at AliExpress e.g. the HT7833 with the long pins.

**Reply by OberfragGER** [reply]
> Replacement is a quite tricky operation because it is very tiny on the board. But you can easily install a second one between the 5V, 3.3V and ground connector. This works very well. You might order at AliExpress e.g. the HT7833 with the long pins.

@stefanbode Sorry for my late question. how do i wire the pins? HT7833 has 3 Pins. But one which the Vin and which one is vout? Vin = 5V, Vout =3V???

**Reply by stefanbode** [reply]
You have to made a Google search for the pinout. Normally easy to find

**Reply by OberfragGER** [reply]
> You have to made a Google search for the pinout. Normally easy to find

I already did so! Did´nt found something useful. The use of HT7833 seems to be very rare.

**Reply by sfromis** [reply]
Sounds strange. When Googling for HT7833, the very first result had a link to a data sheet with: 
![image](https://github.com/user-attachments/assets/a3fe5ee6-c607-4a54-a984-d1be41ba4fd2)

**Reply by OberfragGER** [reply]
I have a "chip" simmilar to that on on the Wemos. but i cant read the markings on top (too dark, too small).
And then just put that one on top?
![image](https://github.com/user-attachments/assets/95d8b61a-aebd-491d-a2e2-dba63e23ef9a)
But in the picture i have 3 legs instead of 5...
So just put some wires between...? Like this?
![image](https://github.com/user-attachments/assets/552c6ed3-2128-4181-8cc1-2976a23278f8)

**Comment by vinceg0267** [comment]
> Replacement is a quite tricky operation because it is very tiny on the board. But you can easily install a second one between the 5V, 3.3V and ground connector. This works very well. You might order at AliExpress e.g. the HT7833 with the long pins.

No, it's not my forte operating on these small circuits. I think I explained myself badly. I intended to exchange the Wemos D1 for a decent one.
Jason was talking about NodeMCU, I saw the Lolin D1, do you think it's good? Does it deliver 500mA on the 3.3V output? NodeMCU I don't know what they are, what's the difference?

**Reply by barbudor** [reply]
If you are on changing for another board, I strongly suggest going for a board based on ESP32-C3
Not much more expansive and much more powerfull
For example:
https://a.aliexpress.com/_EyuAElt
https://a.aliexpress.com/_EzVPPZV

**Reply by vinceg0267** [reply]
I have a pair of Esp32 wrom32s, can I use one of these instead of the wemos?

**Reply by sfromis** [reply]
Dev boards based on ESP32-WROOM are like NodeMCU, just with ESP32 instead of ESP8266. Should generally be fine, as they are well supported by Tasmota.

**Comment by vinceg0267** [comment]
Ok, I was able to replace the Wemos D1 with an Esp32 board. The three Pzem-004T sensors connected to the board work well.
I wonder if I can connect two more Pzem on the board or is it better to configure another Esp32. What do you think?

**Reply by barbudor** [reply]
On ESP32 you can get up to 8 PZEM
Then the choice is yours, what is more convenient : more wires or more wireless devices?

**Reply by vinceg0267** [reply]
I'm not an expert in these things and I couldn't answer. I'm tempted to say that it is more convenient to have more wires but perhaps the card would be overloaded? On the other hand, adding a wireless device to the router shouldn't cause any problems... what would you do?

**Reply by sfromis** [reply]
From my viewpoint, long wires can be badly inconvenient to mess with. Thus, it depends on the practicality of handling the wires. I'd not expect an ESP32 to be anywhere near overloaded by 8 PZEM devices.

**Comment by vinceg0267** [comment]
I replaced the Wemos D1 Mini with an Esp32, I no longer have reboots. Thank you all.

**Comment by vinceg0267** [comment]
> On ESP32 you can get up to 8 PZEM Then the choice is yours, what is more convenient : more wires or more wireless devices?

I'm reopening this thread for an update. I installed 3 Pzem using esphome and everything works fine. So I thought about adding more Pzem to the same Esp32 board but as soon as I add the fourth one only two work. If I leave three they all work. So I have to think that the Esp32 board doesn't support more than three Pzem or is the problem EspHome?

**Reply by Jason2866** [reply]
This is Tasmota Discussions. If you have questions regarding esphome ask there.

**Reply by barbudor** [reply]
And the discussion is about hardware watchdog, not PZEM

**Comment by jdm-twtr** [comment]
is there any documentation on the voltage regulator issue? My D1 mini clone appears to have a DE=A1D marking, suggesting it is a WL9003S5-33 300mA linear voltage regulator. My 12VDC power supply provides 750mA of current, and reads at approximately 14V. I'm using a DC power shield and relay shield, so the D1 and relay are both powered from the 5V rail and should not be drawing current. My design is such that the relay is configured to always be ON but is wired in the NO position and tasmota default poweron state is ON. I have a switch that is NC and pushing it opens the circuit, triggering the relay to turn off. So the switch being NC probably has a negligible current draw. I still think that the 300mA would be sufficient.

The restarts only appear to happen when the relay (powering a magnetic door lock, also energized via the 12V 750mA supply) re-engages ONLY in the case where the door is shut. If the relay engages while the door is still open, the restart does not happen.

My only suspicion here is that the power supply is not providing clean enough power. The DC adapter appears to read closer to 14V when measured by a DMM. 

Do you still suspect the onboard regulator to be a problem in this case?

**Reply by jdm-twtr** [reply]
Oh -- current measures around 300mA on the whole setup when leads are in series with the ground wire back to the supply.

**Reply by jdm-twtr** [reply]
that's MAX current through an entire lock engage/disengage cycle.

**Reply by Jason2866** [reply]
That's the problem `WL9003S5-33 300mA`.  300mA is too low. Needs replacing against a 500mA type (as the orig. D1 mini do have).

**Reply by AleXSR700** [reply]
Would this also be a solution for Shelly devices like 1PM, 2.5 etc.? (the non-plus versions)

I had a few watchdog restarts on a 2.5 and when searching for reasons, I found this thread.

**Reply by sfromis** [reply]
That would depend on what voltage regulator is already on the boards. When it comes to Shelly 2.5, I think I saw someone report better stability after replacing a capacitor gone bad.

**Reply by AleXSR700** [reply]
Yes, one of the Elkos died on me. Will need to check if that is the only issue.
